scripts/ci/tools/freespace.py

```diff
@@ -0,0 +1,68 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""freespace.py for cean environment before to start CI"""
+
+from platform import version
+import shlex
+import subprocess
+import click
+from click import ClickException
+from rich.console import Console
+from typing import List
+
+console = Console(force_terminal=True, color_system="standard", width=180)
+
+option_verbose = click.option(
+    "--verbose",
+    is_flag=True,
+    help="Print verbose information about free space steps",
+)
+
+@click.group()
+def main():
+    pass
+
+@option_verbose
+def run_command(cmd: List[str], verbose, *, check: bool = True, **kwargs):
+    if verbose:
+        #print(f"$ {' '.join(shlex.quote(c) for c in cmd)}")
```

Comment:
Why not  using the shlex quote? It's actually better, becase it will add "" around parameters if they have spaces potentially.

Reply:
Yeah, I thought rich (console.print() ) doesn't work with shlex, do you have reason, I will fix it

Reply:
It will work. Shlex is only to format the string.